src/text-expander-element.ts

```diff
@@ -187,7 +187,7 @@ class TextExpander {
   }
 
   onKeydown(event: KeyboardEvent) {
-    if (event.key !== 'Escape') return
+    if (event.key !== 'Escape' && (!this.menu || !this.combobox)) return
     this.deactivate()
     event.stopImmediatePropagation()
     event.preventDefault()
```

Comment:
I think it might be better to flip this whole conditional now, to make it read easier:

```suggestion
    if (event.key === 'Escape' && (this.menu || this.combobox)) {
      this.deactivate()
      event.stopImmediatePropagation()
      event.preventDefault()
    }
```

Reply:
true 😅